UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarnings/Admin/RadioNavigationalWarningCreateAdminTest.cs

```diff
@@ -64,12 +73,75 @@ public async Task WhenCallAddRadioNavigationalWarnings_ThenNewRecordIsCreated()
+        {
+            _controller.TempData = _tempData;
+            const string expectedView = "~/Views/RadioNavigationalWarningsAdmin/Create.cshtml";
+            DefaultHttpContext httpContext = new();
+            FormCollection formCol = new(new Dictionary<string, StringValues>
+                                        {
+                                            {"SkipDuplicateReferenceCheck", "No" }
+                                        });
+            httpContext.Request.Form = formCol;
+            _controller.ControllerContext.HttpContext = httpContext;
+
+            RadioNavigationalWarning radioNavigationalWarning = GetFakeRadioNavigationalWarning();
+            radioNavigationalWarning.Id++;
+            radioNavigationalWarning.IsDeleted = false;
+
+            await SeedWarningType(GetFakeWarningTypes());
+            await SeedRadioNavigationalWarnings(new List<RadioNavigationalWarning>() { radioNavigationalWarning });
+
+            IActionResult result = await _controller.Create(radioNavigationalWarning);
+
+            Assert.IsInstanceOf<IActionResult>(result);
+            Assert.AreEqual("A warning record with this reference number already exists. Would you like to add another record with the same reference?", _controller.TempData["message"].ToString());
+            Assert.IsInstanceOf<ViewResult>(result);
+            string actualView = ((ViewResult)result).ViewName;
+            Assert.AreEqual(expectedView, actualView);
+            Assert.IsTrue(((ViewResult)result).ViewData.ModelState.IsValid);
+        }
+
+        [Test]
+        public async Task WhenCallCreateWithExistingReferenceNumberAndSkipDuplicateReferenceCheckFlagAsYes_ThenNewRecordIsCreated()
+        {
+            _controller.TempData = _tempData;
+            DefaultHttpContext httpContext = new();
+            FormCollection formCol = new(new Dictionary<string, StringValues>
+                                        {
+                                            {"SkipDuplicateReferenceCheck", "Yes" }
+                                        });
+            httpContext.Request.Form = formCol;
+            _controller.ControllerContext.HttpContext = httpContext;
+
+            RadioNavigationalWarning radioNavigationalWarningNewEntry = GetFakeRadioNavigationalWarning();
+            radioNavigationalWarningNewEntry.Id += 2;
+            radioNavigationalWarningNewEntry.IsDeleted = false;
+
+            IActionResult result = await _controller.Create(radioNavigationalWarningNewEntry);
+
+            Assert.IsInstanceOf<IActionResult>(result);
+            Assert.AreEqual("Record created successfully!", _controller.TempData["message"].ToString());
+            Assert.AreEqual("Index", ((RedirectToActionResult)result).ActionName);
+            Assert.AreEqual(3, FakeContext.RadioNavigationalWarnings.ToListAsync().Result.Count);
```

Comment:
After some investigation, I now realise the 2 added to the ID is because it assumes the previous test has ran and added an entry (the previous test increments the ID by 1), so there is a dependency on the previous test running for this to succeed.

I run my tests in NCrunch, which runs all tests in isolation and is therefore why this was failing. Running the tests through Visual Studio's test explorer and they do all work BUT ONLY IF you run all the tests in the test class.

I have discussed this with Neville, who despite having approved this PR agrees that tests should run in isolation and so this needs addressing please.

Each test should be able to run on its own, in isolation, without depending on another test. So you could either use the SetUp to create a share list of RNWs, but then you would have to add the duplicates in the test itself.

Hope that makes sense! Happy to discuss if not.

Reply:
thanks for highlighting , done required changes.